Battleships/Program.cs

```diff
@@ -0,0 +1,343 @@
+    }
+
+    Horizontal:
+    Console.WriteLine("Horizontal (y/n)?");
+    string inputHorizontal = Console.ReadLine();
+    if (inputHorizontal != "y" && inputHorizontal != "n")
+    {
+        Console.WriteLine("Invalid input, only 'y'(yes) or 'n'(no) are valid inputs");
+        goto Horizontal;
+    }
+
+    bool isHorizontal = inputHorizontal == "y";
+    if (!IsViablePlacement(ship, isHorizontal, playerGrid[player], row, collumn))
+    {
+        Console.WriteLine("Invalid Placement, must be placed 2 units away from other ships and inside the grid");
+        goto PlaceShip;
+    }
+
+    for (int i = 0; i < ship.size; i++)
+    {
+        playerGrid[player].position[row, collumn].ship = ship;
+
+        if (inputHorizontal == "y")
+            collumn++;
+        else
+            row++;
+    }
+}
+
+bool IsViablePlacement(Ship ship, bool isHorizontal, Grid playerGrid, int row, int collumn)
+{
+    for (int i = 0; i < ship.size; i++)
+    {
+        // if (playerGrid.position[row, collumn].ship != null)
+        //     return false;
+        if (row >= rows || collumn >= collumns)
+            return false;
+
+        for (int r = -1; r < 2; r++)
+        {
+            if(r == -1 || r == 1)
+                if(row + r < 0 || row + r >= rows)
+                    continue;
+                
+            for (int c = -1; c < 2; c++)
+            {
+                if(c == -1 || c == 1)
+                    if(collumn + c < 0 || collumn + c >= collumns)
+                        continue;
+
+                if (playerGrid.position[row + r, collumn + c].ship != null && playerGrid.position[row + r, collumn + c].ship != ship)
+                    return false;
+            }
+        }
+
+
+        if (isHorizontal)
+            collumn++;
+        else
+            row++;
+    }
+
+    return true;
+}
+
+
+
+void InputPosition(out int inputCollumn, out int inputRow)
+{
+    InputPos:
+    string placementPosition = Console.ReadLine();
+
+    if (placementPosition.Length != 2)
+    {
+        Console.WriteLine("Invalid Position, positions are written with a letter(collumn) and a number(row). Ex: 'a0'");
+        goto InputPos;
+    }
+
+    inputCollumn = char.ToUpper(placementPosition[0]) - 65; //Change Letter To a value based on it's alphabetical position
+    inputRow = char.ToUpper(placementPosition[1]) - 48; //Change string value to it's int value
+
+    if (!IsViableGridPosition(placementPosition, inputCollumn, inputRow))
+    {
+        Console.WriteLine(
+            "Invalid Position, positions are written with a letter(collumn) and a number(row). Ex: 'a0'");
+        goto InputPos;
+    }
+}
+
+bool IsViableGridPosition(string inputString, int inputCollumn, int inputRow)
+{
+    if (inputCollumn < 0 || inputCollumn > collumns)
+        return false;
+    if (inputRow > rows)
+        return false;
+
+    return true;
+}
+
+
+void DrawGameBoard(int player, int otherPlayer, bool placementPhase)
+{
+    string collumnLine = " | ";
```

Comment:
Technically, it'd be smarter to use a `StringWriter` to avoid unnecessary allocations here. But it shouldn't affect your game! :D